javascript/MaterialXView/source/index.js

```diff
@@ -4,484 +4,116 @@
-        new Promise(resolve => hdrloader.setDataType(THREE.FloatType).load('Lights/san_giuseppe_bridge_split.hdr', resolve)),
-        new Promise(resolve => fileloader.load('Lights/san_giuseppe_bridge_split.mtlx', resolve)),
-        new Promise(resolve => hdrloader.setDataType(THREE.FloatType).load('Lights/irradiance/san_giuseppe_bridge_split.hdr', resolve)),
-        new Promise(resolve => loadGeometry(scene, gltfLoader, geometryFilename, resolve)),
+        new Promise(resolve => hdrLoader.setDataType(THREE.FloatType).load('Lights/san_giuseppe_bridge_split.hdr', resolve)),
+        new Promise(resolve => fileLooder.load('Lights/san_giuseppe_bridge_split.mtlx', resolve)),
+        new Promise(resolve => hdrLoader.setDataType(THREE.FloatType).load('Lights/irradiance/san_giuseppe_bridge_split.hdr', resolve)),
         new Promise(function (resolve) {
             MaterialX().then((module) => {
                 resolve(module);
             });
-        }),
-        new Promise(resolve => materialFilename ? fileloader.load(materialFilename, resolve) : resolve())
-
-    ]).then(async ([loadedRadianceTexture, loadedLightSetup, loadedIrradianceTexture, { scene: obj }, mxIn, mtlxMaterial]) => {
-
-        // Initialize MaterialX and the shader generation context
-        mx = mxIn;
-        let doc = mx.createDocument();
-        let gen = new mx.EsslShaderGenerator();
-        let genContext = new mx.GenContext(gen);
-        let stdlib = mx.loadStandardLibraries(genContext);
-        doc.importLibrary(stdlib);
-
-        // Set search path.
-        const searchPath = 'Materials/Examples/StandardSurface/';
+        }) 
+    ]).then(async ([loadedRadianceTexture, loadedLightSetup, loadedIrradianceTexture, mxIn]) => 
+    {
+        // Initialize viewer + lighting
+        await viewer.initialize(mxIn, renderer, loadedRadianceTexture, loadedLightSetup, loadedIrradianceTexture);
 
         // Load material
-        if (mtlxMaterial)
-            await mx.readFromXmlString(doc, mtlxMaterial, searchPath);
-        else
-            fallbackMaterial(doc);
-
-        // Search for any renderable items
-        let elem = mx.findRenderableElement(doc);
-
-        // Load lighting setup into document
-        const lightRigDoc = mx.createDocument();
-        await mx.readFromXmlString(lightRigDoc, loadedLightSetup);
-        doc.importLibrary(lightRigDoc);
-
-        // Register lights with generation context
-        const lights = findLights(doc);
-        const lightData = registerLights(mx, lights, genContext);
-
-        const radianceTexture = prepareEnvTexture(loadedRadianceTexture, renderer.capabilities);
-        const irradianceTexture = prepareEnvTexture(loadedIrradianceTexture, renderer.capabilities);
-
-        const threeMaterial = generateMaterial(elem, gen, genContext, lights, lightData,
-            textureLoader, radianceTexture, irradianceTexture, gui);
+        viewer.getMaterial().loadMaterial(viewer, materialFilename);
```

Comment:
As much as possible has been hidden now in the viewer class. The main calls are
loading a material or loading geometry, which are used in the callbacks when menu selection changes as well as here on page load.